### PR TITLE
feat(tab): Convert JS to TypeScript

### PR DESCRIPTION
**Pull request by acdvorak**
Refs #4225

### CODE REVIEW

File: packages/mdc-tab/index.ts [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+import {MDCComponent} from '@material/base/component';
+import {MDCRipple, MDCRippleFoundation, RippleCapableSurface} from '@material/ripple/index';
+import {MDCTabIndicator} from '@material/tab-indicator/index';
```
Comment (acdvorak) [inline packages/mdc-tab/index.ts]: Depends on PR #4391

**Comment by moog16** [comment]
are there supposed to be other commits? Not much for the tab package in this PR.

**Comment by acdvorak** [comment]
Woops, looks like I missed some Closure syntax in the README. Fixed.

**Comment by moog16** [comment]
oh thanks for catching that I haven't been checking the readmes

File: packages/mdc-tab/index.ts [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+
+  id!: string; // assigned in initialize();
+
+  // Public visibility for this property is required by RippleCapableSurface.
+  root_!: HTMLElement; // assigned in MDCComponent constructor
+
+  private ripple_!: MDCRipple; // assigned in initialize();
+  private tabIndicator_!: MDCTabIndicator; // assigned in initialize();
+  private content_!: HTMLElement; // assigned in initialize();
+  private handleClick_!: EventListener; // assigned in initialize();
```
Comment (moog16) [inline packages/mdc-tab/index.ts]: typed to SpecificEventListener<'click'>

also assigned in initialSyncWithDom

File: packages/mdc-tab/index.ts [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+  }
+
+  set focusOnActivate(focusOnActivate: boolean) {
+    this.foundation_.setFocusOnActivate(focusOnActivate);
+  }
+
+  /**
+   * Activates the tab
+   */
+  activate(computeIndicatorClientRect?: ClientRect | null) {
```
Comment (moog16) [inline packages/mdc-tab/index.ts]: in tab-indicator we changed this to just `computeIndicatorClientRect?: ClientRect`

File: packages/mdc-tab/types.ts [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+ * the visual position of the Tab with respect it's parent container.
+ */
+export interface MDCTabDimensions {
+  rootLeft: number;
+  rootRight: number;
+  contentLeft: number;
+  contentRight: number;
+}
+
+// TODO(acdvorak): Use this style everywhere
```
Comment (moog16) [inline packages/mdc-tab/types.ts]: do you need this comment anymore?

File: packages/mdc-tab/index.ts (hunk 2 above)
Comment (acdvorak) [reply packages/mdc-tab/index.ts]: Done

File: packages/mdc-tab/index.ts (hunk 3 above)
Comment (acdvorak) [reply packages/mdc-tab/index.ts]: Done

File: packages/mdc-tab/types.ts (hunk 4 above)
Comment (acdvorak) [reply packages/mdc-tab/types.ts]: Removed

**Pull request merged**